### PR TITLE
Improve navigation add Quick Start intro

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR:
 1. Improves the navigation.
 2. Removes dummy pages.
 3. Improves Validation Guide.
 4. Adds introduction text for the Quick Start guide. This guide is incomplete. More text will be added in following PRs.

### CODE REVIEW

**Comment by alexander-yevsyukov** [comment]
@armiol @dmdashenkov 
Guys, even though the new text of Quick Start is not complete, I believe it would be better to push this PR sooner as it improves the structure and removes the dead pages.

File: docs/client-libs/dart.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+title: Dart Client Library
+headline: Documentation
+bodyclass: docs
+layout: docs
+---
+# Dart Client Library
```
Comment (dmdashenkov) [inline docs/client-libs/dart.md:7]: Please add references to repositories/generated documentation for client libraries here and on Java and JS pages as well.

File: docs/examples/index.md [hunk 2]
```diff
@@ -4,15 +4,15 @@ headline: 'Spine Examples'
 bodyclass: 'docs'
 layout: docs
 ---
-<p>Example are available through the <a target="_blank" href="https://github.com/spine-examples">Spine Examples</a> GitHub organization.  
+<p class="lead">Example are available through the <a target="_blank" href="https://github.com/spine-examples">Spine Examples</a> GitHub organization.
```
Comment (dmdashenkov) [inline docs/examples/index.md:7]:
```suggestion
<p class="lead">Examples are available through the <a target="_blank" href="https://github.com/spine-examples">Spine Examples</a> GitHub organization.  
```

File: docs/guides/validation-user-guide.md [hunk 3]
```diff
@@ -60,24 +72,35 @@ ValidationError error = validate(msg);
 Similarly to `validate()` method in Java, the validation function does not throw exceptions. A list
 of `ConstraintViolation`s can be obtained from the `ValidationError`.
 
-## Validation options
-### Required fields
+## Validation options overview
+<a name="validation-options"></a>
+
+In most cases validation constraints are defined for Protobuf message fields such as if a field
+must be populated or its must be withing a range, or have match a regular expression. Not so often
```
Comment (dmdashenkov) [inline docs/guides/validation-user-guide.md:79]:
```suggestion
must be populated or it must be withing a range, or match a regular expression. Not so often
```

File: docs/quickstart-java.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+title: Getting Started in Java
+headline: Documentation
+bodyclass: docs
+layout: docs
+---
+
+[//]: <> (The base path for code samples is `_samples/examples/src/main/`.)
+[//]: <> (Change it to `_examples/hello/`.)
+
+<?code-excerpt path-base="../../../../_examples/hello/"?>
```
Comment (dmdashenkov) [inline docs/quickstart-java.md:11]: Seems like we aren't using code excepts in this file. Why change the `path-base`?

Comment (alexander-yevsyukov) [reply docs/quickstart-java.md:11]: I am going to use it later.

File: docs/guides/rejections.md [hunk 5]
```diff
@@ -6,23 +6,22 @@ layout: docs
...
 differently. Here is the summary of the differences:
 
-* `java_multiple_files file` option must be set to `false`
+1. `java_multiple_files` file option must be set to `false`
 
-By doing so we instruct Protobuf Compiler to put all the rejection classes in one outer class.
-Spine Model Compiler for Java generates `ThrowableMessage` classes for all these messages. 
-These classes will be named after the classes of rejection messages.
-Putting rejection message classes under an outer class avoids name clash inside the package.
+    By doing this we instruct Protobuf Compiler to put all the rejection classes in one outer class.
```
Comment (armiol) [inline docs/guides/rejections.md:14]: `...into a single outer class.`

File: docs/introduction/index.md [hunk 6]
```diff
@@ -1,26 +1,255 @@
...
+
+It is likely that some of the bits of this picture would change during the process.
+But the whole team, including domain experts, need to have complete understanding of how the 
+business works to avoid “surprises” down the road. 
+
+We return to learning the domain when we discover inconsistencies in the model,
+or we need more information about how the business works, or the business wants to develop further
+and we need to update the model.
+
+Once we got enough domain knowledge we proceed to the implementation.
```
Comment (armiol) [inline docs/introduction/index.md:37]: Do we want to mention #EventStorming before this sentence?

File: docs/introduction/index.md [hunk 7]
```diff
@@ -1,26 +1,255 @@
...
+        builder().setWhenDone(ctx.getTimestamp());
+    }
+}
+```
+
+#### Repositories
+The framework provides default implementations for repositories.
+A custom `Repository` class may be needed for:
+  * <strong>Dispatching messages to entities in a non-standard way</strong>.
+    By default, a command is dispatched using the first field of the command message.
```
Comment (armiol) [inline docs/introduction/index.md:154]: `...using the first field of the command message as an ID of the target entity.`

File: docs/introduction/index.md [hunk 8]
```diff
@@ -1,26 +1,255 @@
...
+    An event is dispatched by the ID of the entity which generated the event.
```
Comment (armiol) [inline docs/introduction/index.md:155]: I would say `emitted the event.`

File: docs/client-libs/dart.md (hunk 1 above)
Comment (alexander-yevsyukov) [reply docs/client-libs/dart.md:7]: It's a good idea! Added.

File: docs/introduction/index.md (hunk 6 above)
Comment (alexander-yevsyukov) [reply docs/introduction/index.md:37]: We already mention in when we start the domain knowledge section. There's the link in the first paragraph of the section.

**Comment by alexander-yevsyukov** [comment]
@dmdashenkov, I've addressed your request, and will “dismiss” your review now as I need to push the changes.

**Pull request merged**